interfaceUtils.py

```diff
@@ -40,127 +46,139 @@ def requestBuildArea(self):
                 z1 = buildArea["zFrom"]
                 x2 = buildArea["xTo"]
                 z2 = buildArea["zTo"]
-                # print(buildArea)
                 buildArea = (*self.global2local(x1, None, z1),
                              *self.global2local(x2 - x1, None, z2 - z1))
             return buildArea
         else:
             print(response.text)
-
-    def runCommand(self, command):
-        """**Runs a Minecraft command in the world.**
-
-        TODO: extract and convert local coordinates
-        """
-
-        # TODO: local2global
-        print("WARNING: Interface.runCommand not correctly implemented yet!")
-
-        # DEBUG: print("running cmd " + command)
-        url = 'http://localhost:9000/command'
-        try:
-            response = requests.post(url, bytes(command, "utf-8"))
-        except ConnectionError:
-            return "connection error"
-
-        # TODO: global2local
-
-        return response.text
+            return -1
 
     def getBlock(self, x, y, z):
-        """**Returns the name of a block in the world.**"""
+        """**Return the name of a block in the world**."""
         x, y, z = self.local2global(x, y, z)
 
         url = 'http://localhost:9000/blocks?x={}&y={}&z={}'.format(x, y, z)
-        # DEBUG: print(url)
         try:
             response = requests.get(url)
         except ConnectionError:
             return "minecraft:void_air"
         return response.text
-        # DEBUG: print("{}, {}, {}: {} - {}".format(x, y, z, response.status_code, response.text))
 
-    def setBlock(self, x, y, z, blockStr):
-        if self.useBatching:
-            self._placeBlockBatched(x, y, z, blockStr, self.batchSize)
+    def fill(self, x1, y1, z1, x2, y2, z2, str):
+        x1, y1, z1 = self.local2global(x1, y1, z1)
+        x2, y2, z2 = self.local2global(x2, y2, z2)
+        xlo, ylo, zlo = min(x1, x2), min(y1, y2), min(z1, z2)
+        xhi, yhi, zhi = max(x1, x2), max(y1, y2), max(z1, z2)
+
+        for x in range(xlo, xhi + 1):
+            for y in range(ylo, yhi + 1):
+                for z in range(zlo, zhi + 1):
+                    self.setBlock(x, y, z, str)
+
+    def setBlock(self, x, y, z, str):
+        """**Place a block in the world depending on buffer activation**."""
+        if self.__buffering:
+            self.placeBlockBatched(x, y, z, str, self.bufferlimit)
         else:
-            self._setBlockDirect(x, y, z, blockStr)
-
-    def sendOutstandingBlocks(self):
-        self._sendBlocks()
+            self.placeBlock(x, y, z, str)
 
-    def _setBlockDirect(self, x, y, z, blockStr):
-        """**Places a block in the world.**"""
+    def placeBlock(self, x, y, z, str):
```

Comment:
I renamed this to blockStr intentionally, because str is an internal function

Reply:
Fair point, will amend.